app/services/request_mentoring_on_solution.rb

```diff
@@ -6,6 +6,7 @@ class RequestMentoringOnSolution
   def call
     # We guard this for exercises that are promoted to core
     unless (user_track.mentored_mode? && solution.exercise.core?)
+      return unless solution.track.accepting_new_students?
```

Comment:
Should we wrap this in `Solution#allowing_mentors?` (or something like this) for looser coupling?

Reply:
I'll add a commit. Again, the prefix of "track" is important to me here, so it would be `Solution#track_allowing_mentors?`.